go/pserver/optimizer.go

```diff
@@ -35,7 +35,7 @@ func cArrayToSlice(p unsafe.Pointer, len int) []byte {
 	return (*[1 << 30]byte)(p)[:len:len]
 }
 
-func newOptimizer(paramWithConfigs ParameterWithConfig, State []byte) *optimizer {
+func newOptimizer(paramWithConfigs *ParameterWithConfig, State []byte) *optimizer {
```

Comment:
Is passing pointer `* ParameterWithConfig` necessary? Please see: https://github.com/golang/go/wiki/CodeReviewComments#pass-values